editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -45,6 +46,55 @@
 #include "scene/gui/texture_rect.h"
 #include "scene/gui/tree.h"
 
+void HighlightedLabel::draw_substr_rects(const Vector2i &p_substr, Vector2 p_offset, const int p_line_limit, const int line_spacing) {
+	for (int i = get_lines_skipped(); i < p_line_limit; i++) {
+		RID line = get_line_rid(i);
+		Vector<Vector2> ranges = TS->shaped_text_get_selection(line, p_substr.x, p_substr.x + p_substr.y);
+		Rect2 line_rect = get_line_rect(i);
+		for (const Vector2 &range : ranges) {
+			Rect2 rect = Rect2(Point2(range.x, 0) + line_rect.position, Size2(range.y - range.x, line_rect.size.y));
+			rect.position = p_offset + line_rect.position;
+			rect.position.x += range.x;
+			rect.size = Size2(range.y - range.x, line_rect.size.y);
+			rect.size.x = MIN(rect.size.x, line_rect.size.x - range.x);
+			if (rect.size.x > 0) {
+				draw_rect(rect, Color(1, 1, 1, 0.07), true);
+				draw_rect(rect, Color(0.5, 0.7, 1.0, 0.4), false, 1);
+			}
+			p_offset.y += line_spacing + TS->shaped_text_get_ascent(line) + TS->shaped_text_get_descent(line);
```

Comment:
```suggestion
void HighlightedLabel::draw_substr_rects(const Vector2i &p_substr, const Vector2 &p_offset, int p_line_limit, int p_line_spacing) {
	for (int i = get_lines_skipped(); i < p_line_limit; i++) {
		RID line = get_line_rid(i);
		Vector<Vector2> ranges = TS->shaped_text_get_selection(line, p_substr.x, p_substr.x + p_substr.y);
		Rect2 line_rect = get_line_rect(i);
		for (const Vector2 &range : ranges) {
			Rect2 rect = Rect2(Point2(range.x, 0) + line_rect.position, Size2(range.y - range.x, line_rect.size.y));
			rect.position = p_offset + line_rect.position;
			rect.position.x += range.x;
			rect.size = Size2(range.y - range.x, line_rect.size.y);
			rect.size.x = MIN(rect.size.x, line_rect.size.x - range.x);
			if (rect.size.x > 0) {
				draw_rect(rect, Color(1, 1, 1, 0.07), true);
				draw_rect(rect, Color(0.5, 0.7, 1.0, 0.4), false, 1);
			}
			p_offset.y += p_line_spacing + TS->shaped_text_get_ascent(line) + TS->shaped_text_get_descent(line);
```

Reply:
I'm mutating `p_offset` here for the (unused) multiline case where a highlight substr spans multiple lines.